src/DynamoCoreWpf/Commands/AutomationSettings.cs

```diff
@@ -233,18 +234,23 @@ internal string SaveRecordedCommands()
             XmlElement commandRoot = document.CreateElement("Commands");
             document.AppendChild(commandRoot);
 
+            const string format = "Commands-{0:yyyyMMdd-hhmmss}.xml";
+            string xmlFileName = string.Format(format, DateTime.Now);
+            string xmlFilePath = Path.Combine(Path.GetTempPath(), xmlFileName);
+            
             // Create attributes that applied to the entire recording.
             var helper = new XmlElementHelper(commandRoot);
             helper.SetAttribute(EXIT_ATTRIB_NAME, ExitAfterPlayback);
             helper.SetAttribute(PAUSE_ATTRIB_NAME, PauseAfterPlayback);
             helper.SetAttribute(INTERVAL_ATTRIB_NAME, CommandInterval);
+            // Serialization in SaveContext.File may need file path. Add it
+            // temporarily and remove it after searilization.
+            helper.SetAttribute(Configurations.FilePathAttribName, xmlFilePath);
```

Comment:
Please use the utility method `SetDocumentXmlPath` for doing these, for this case, it should be:

```csharp
Utils.SetDocumentXmlPath(document, xmlFilePath);

foreach (DynamoModel.RecordableCommand command in recordedCommands)
    commandRoot.AppendChild(command.Serialize(document));

Utils.SetDocumentXmlPath(document, null); // Remove file path.
```